src/sagas/projects.js

```diff
@@ -44,6 +47,7 @@ export function* applicationLoaded(action) {
 
 export function* createProject() {
   yield put(projectCreated(generateProjectKey()));
+  yield put(validatedSource('javascript', []));
```

Comment:
So this approach definitely works, but kind of as a second-order effect, because the `compiledProjects` saga [consumes this action and initiates compilation in response to it](https://github.com/alessbell/popcode/blob/dc49520b55c57487166fdeac44aaa1449b41240a/src/sagas/compiledProjects.js#L35).

My preference is to keep action dispatches as close as possible to an unvarnished record of what actually happened (somebody clicked a button, some async operation finished, etc.) In this case we are saying we’ve finished validating a source, but in fact we haven’t (this intuition is strengthened, I think, by the fact that we have to arbitrarily pick a language to say we have validated).

My proposal would be: instead of adding a new dispatch into the flow, simply make the system react more robustly to the actions that are already being dispatched. In particular, I think the `compiledProjects` saga should consume the `PROJECT_CREATED` action and compile the project in response.

LMK what you think!

Reply:
Yeah, makes sense! This solution indeed felt slightly wrong with the language hard-coded. 

The thinking was that, given that the source is a brand new Popcode, it can safely be assumed valid. I was after that second-order effect :) Much clearer to simply compile the project, updated.